nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/MemoryWorkQueuing.java

```diff
@@ -86,28 +87,23 @@ public void workReschedule(String queueId, Work work) {
     }
 
     Optional<Work> lookup(String workId) {
-        return allQueued.values().stream().map(queue -> queue.lookup(workId)).filter(work -> work != null).findAny();
+        return allQueued.values().stream().map(queue -> queue.lookup(workId)).filter(Objects::nonNull).findAny();
     }
 
     @Override
     public Work find(String workId, State state) {
-        return lookup(workId)
-                .filter(work -> work.getWorkInstanceState() == state)
-                .orElse(null);
+        return lookup(workId).filter((work) -> {
+            State workState = work.getWorkInstanceState();
+            return state == null ? (workState == State.SCHEDULED || workState == State.RUNNING) : workState == state;
+        }).orElse(null);
```

Comment:
As it's used several times you could put this logic in a small static method and do:

    return lookup(workId).filter(work -> workHasState(work, state)).orElse(null);

    /** Appropriate Javadoc here... */
    protected static boolean workHasState(Work work, State state) {
        State workState = work.getWorkInstanceState();
        return state == null ? (workState == State.SCHEDULED || workState == State.RUNNING) : workState == state;
    }

Reply:
Sure, where would I put such a method? As static and independent from the work queue, `MemoryWorkQueuing` doesn't seem right.

Reply:
It's a small helper method to avoid repetition, you can put it in this class without problem. If we find a better place later we'll move it :)

Reply:
:+1: